KubeArmor/core/kubeArmor.go

```diff
@@ -448,6 +469,64 @@ func KubeArmor() {
 
 	// == //
 
+	if !dm.K8sEnabled && cfg.GlobalCfg.Policy {
+		// check if the CRI socket set while executing kubearmor exists
+		if cfg.GlobalCfg.CRISocket != "" {
+			trimmedSocket := strings.TrimPrefix(cfg.GlobalCfg.CRISocket, "unix://")
+			if _, err := os.Stat(trimmedSocket); err != nil {
+				dm.Logger.Warnf("Error while looking for CRI socket file: %s", err.Error())
+
+				// destroy the daemon
+				dm.DestroyKubeArmorDaemon()
+				return
+			}
+		}
+
+		if hostContainerRuntime() == "docker" {
+			if cfg.GlobalCfg.CRISocket == "" {
+
+				socketFile := kl.GetCRISocket("docker")
+				if socketFile != "" {
+					cfg.GlobalCfg.CRISocket = "unix://" + socketFile
+				}
+			}
+			// update already deployed containers
+			dm.GetAlreadyDeployedDockerContainers()
+
+			// monitor docker events
+			go dm.MonitorDockerEvents()
+
+		} else if hostContainerRuntime() == "containerd" {
+			if cfg.GlobalCfg.CRISocket == "" {
+
+				socketFile := kl.GetCRISocket("containerd")
+
+				if socketFile != "" {
+					cfg.GlobalCfg.CRISocket = "unix://" + socketFile
+				}
+			}
+			// monitor containerd events
+			go dm.MonitorContainerdEvents()
+
+		} else if hostContainerRuntime() == "crio" {
+			if cfg.GlobalCfg.CRISocket == "" {
+				socketFile := kl.GetCRISocket("crio")
+
+				if socketFile != "" {
+					cfg.GlobalCfg.CRISocket = "unix://" + socketFile
+				}
+			}
+			// monitor cri-o events
+			go dm.MonitorCrioEvents()
+
+		} else if hostContainerRuntime() == "unavailable" {
```

Comment:
Note that with this check if the k8s is disabled and if the policy is enabled but there is no container runtime available then the kubearmor will quit. Imo, kubearmor should still continue and apply policies at the host level and reject container based policies.

Reply:
Okay, makes sense. I will update it.